homeassistant/components/honeywell/sensor.py

```diff
@@ -0,0 +1,150 @@
+"""Support for Honeywell (US) Total Connect Comfort climate systems sensors."""
+
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_REGION,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+from . import CLIENT_KEY_COORDINATOR, HoneywellDevice
+
+CONF_DEV_ID = "thermostat"
+CONF_LOC_ID = "location"
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = vol.All(
+    cv.deprecated(CONF_REGION),
+    PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept YAML configuration for integrations that connect to devices or services.

Instead, a configuration flow via the UI should be used.

For more information about this decision see:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>
- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

To get started on the integration configuration, see our developer documentation:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

Reply:
Thanks for getting back to me. I think for this change I will try using discovery.load_platform instead of modifying the yaml structure.
`hass.helpers.discovery.load_platform("sensor", "honeywell", {}, config)`
Would that work here?

I would be open to converting this to use config entries in the future, but at this point I think that's too big of a change for me.

Reply:
We don't allow new platform schemas. This needs a config flow and a major refactor. So unfortunately it'll be quite a big task.

If you don't have time for that, let's close this PR.

If you do want to work on config flow, one way of allowing more time for it, is to break up the PR into smaller pieces and use a separate target branch instead of dev branch, to allow multiple PRs before releasing everything together.

Please let us know your intentions.